src/features/staking/uils/useCalculateTokenProportions.ts

```diff
@@ -0,0 +1,12 @@
+export const useCalculateTokenProportions = (
```

Comment:
Another subjective point, but I've found it nicer to use objects when you have a number of similar inputs (like 3 numbers in this case):

i.e.
`useCalculateTokenProportions({ unstakable, staked, stakable } : { unstakable:string, staked:string, stakable:string })`

Other notes on this function:

1. Why are you using strings? I assumed it was to accept bignumbers, but if you need bignumbers then you can't do `Number()` because it may overflow.
2. This is not really a hook, because it doesn't use any other hooks, it's just a utility function. If you memoized it (you don't need to, just an example), then it would make sense to have it use the hook naming convention (`useXXX`), but as-is it doesn't need to be one. What you could do however, is fetch all the balances in here using `useBalance` and `useContractRead` to work out how much is unstakable, and call the hook `useStakingBalances` or something similar.

Reply:
@gpxl-dev I made a new hook `useTokenBalances` which grabs AST, sAST, and unstakable sAST balances, then I call it in `StakableBar`. Perhaps take another look at it before I squash & merge?